ComponentKit/Core/CKComponent.mm

```diff
@@ -252,6 +265,11 @@ - (CKComponentLayout)computeLayoutThatFits:(CKSizeRange)constrainedSize
                           restrictedToSize:(const CKComponentSize &)size
                       relativeToParentSize:(CGSize)parentSize
 {
+  if (_childComponent) {
+    CKComponentLayout firstChild = [_childComponent layoutThatFits:constrainedSize parentSize:parentSize];
+    return {self, firstChild.size, {{{0,0}, firstChild}}};
+  }
```

Comment:
This is why I'd prefer to keep it in a separate subclass — the cruft here seems unfortunate. Now layoutThatFits has two meanings, depending on whether there is a `childComponent` or not.

Reply:
I agree that this part is not ideal. It's on our to-do list to figure out a better solution.